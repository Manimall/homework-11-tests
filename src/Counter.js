import React, { useState } from "react";

export const Counter = () => {
  const [count, setCount] = useState(0);
  const increment = () => {
    setCount(count => count + 1);
  };

  const decrement = () => {
    setCount(count => count - 1);
  };

  return (
    <>
      <div className="t-count">{count}</div>
      <button className="t-btn-inc" onClick={increment}>Increment</button>
      <button className="t-btn-dec" onClick={decrement}>Decrement</button>
    </>
  );
};
