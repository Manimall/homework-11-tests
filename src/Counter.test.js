// Компонент Counter - простой счётчик
// Напишите для него тесты.
// Убедитесь, что вы протестировали всю функциональнось.
// Также проверьте что компонент рендерится верно, используя Snapshot тест.

// * Задание со звёздочкой - выполнять не обязательно

// Вынесите логику в хук и протестируйте его

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from  'enzyme';
import { Counter } from './Counter';

describe(`Counter`, () => {
	describe(`Component render correctly`, () => {
		it(`renders correctly`, () => {
			const tree = renderer.create(<Counter />).toJSON();
			expect(tree).toMatchSnapshot();
		});
	});

	describe(`Counter correctly`, () => {
		const wrapper = shallow(<Counter />);
		const incBtn = wrapper.find(`.t-btn-inc`);
		const decBtn = wrapper.find(`.t-btn-dec`);
		// const counter = wrapper.find(`.t-count`); // this doesn't not works

		it(`increment correct value`, () => {
			expect(wrapper.find(`.t-count`).text()).toEqual(`0`);
			incBtn.simulate(`click`);
			expect(wrapper.find(`.t-count`).text()).toEqual(`1`);
		});

		it(`decrement correctly`, () => {
			expect(wrapper.find(`.t-count`).text()).toEqual(`1`);
			decBtn.simulate(`click`);
			expect(wrapper.find(`.t-count`).text()).toEqual(`0`);
		});

	});
});
