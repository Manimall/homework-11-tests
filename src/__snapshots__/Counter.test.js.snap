// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component render correctly renders correctly 1`] = `
Array [
  <div
    className="t-count"
  >
    0
  </div>,
  <button
    className="t-btn-inc"
    onClick={[Function]}
  >
    Increment
  </button>,
  <button
    className="t-btn-dec"
    onClick={[Function]}
  >
    Decrement
  </button>,
]
`;
